### PR TITLE
Migrate cluster with(out) quorum js tests as elixir tests

### PR DESCRIPTION
**Pull request by jjrodrig**
## Overview

This PR adds support for testing in different quorum conditions to the elixir test suite. 
The current JS quorum test suite is now included as part of the elixir test suite. 
Two new tasks have been included into Makefile/Makefile.win

```
 make elixir-cluster-without-quorum
``` 
*elixir-cluster-without-quorum*: Executes tests over a cluster without quorum, a 3 node cluster with 2 nodes stopped

```
make elixir-cluster-with-quorum
```
*elixir-cluster-with-quorum*: Executes tests over a cluster with quorum, a 3 node cluster with 1 node stopped

This PR integrates cluster with(out) quorum testing into the current elixir testing infrastructure. The tests to be executed in each quorum condition is selected by using elixir module tags.

```@moduletag :with_quorum_test``` and ```@moduletag :without_quorum_test``` are excluded from the current elixir test suite and are only considered by the associated makefile task.

Any other test that requires to consider cluster quorum conditions can be included by using these tags  

## Testing recommendations

Use the corresponding makefile tasks
```
make elixir-cluster-with-quorum
make elixir-cluster-without-quorum
```

## Related Issues or Pull Requests

PR #1800 include modifications in test/elixir/run and test/elixir/run.cmd used by this PR

## Checklist

- [x] Code is written and works correctly;
- [x] Changes are covered by tests;
- [ ] Documentation reflects the changes;

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -201,19 +201,19 @@ python-black-update: .venv/bin/black
...
 	          --degrade-cluster 2 \
-            --no-eval 'test/elixir/run --only without_quorum_test'
+            --no-eval 'test/elixir/run --only without_quorum_test $(EXUNIT_OPTS)'
 
 .PHONY: elixir-cluster-with-quorum
 elixir-cluster-with-quorum: elixir-check-formatted elixir-credo devclean
 	@dev/run -n 3 -q -a adm:pass \
-						--degrade-cluster 1 \
-						--no-eval 'test/elixir/run --only with_quorum_test'
+			--degrade-cluster 1 \
```
Comment (wohali) [inline Makefile]: Please normalise all of the extra indent to exactly 2 tabs, if you would. (lines 209, 215, etc.)

File: Makefile.win [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+﻿# Licensed under the Apache License, Version 2.0 (the "License"); you may not
```
Comment (wohali) [inline Makefile.win]: What is this change?

Comment (wohali) [reply Makefile.win]: Looks like you inserted a Unicode character at the start of this file. Please revert; it breaks `Makefile.win`.

**Comment by van-mronov** [comment]
@wohali do you mean following output?
```shell
  Software Design
┃
┃ [D] → Found a TODO tag in a comment: # TODO: port sever_port tests from
┃       config.js
┃       test/config_test.exs:73 #(ConfigTest.delete_config)
┃ [D] → Found a TODO tag in a comment: # TODO: In JS we re-fetch _changes with
┃       since_seq, is that
┃       test/replication_test.exs:587 #(ReplicationTest.run_since_seq_repl)
┃ [D] → Found a TODO tag in a comment: # TODO: Parameterize these
┃       test/replication_test.exs:9 #(ReplicationTest)
┃ [D] → Found a TODO tag in a comment: # TODO: switch this to _local when that's
┃       landed
┃       test/config_test.exs:12 #(ConfigTest)
┃ [D] → Found a TODO tag in a comment: # TODO: port remainder of
┃       security_validation.js suite
┃       test/security_validation_test.exs:205 #(SecurityValidationTest)
┃  ...  (5 more, use `-a` to show them)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 3.9 seconds (0.04s to load, 3.9s running checks)
133 mods/funs, found 10 software design suggestions.

Showing priority issues: ↑ ↗ →  (use `--strict` to show all issues, `--help` for options).
```

That's because of `{Credo.Check.Design.TagTODO, [exit_status: 0]}` in our credo config.
If we want to suppress these suggests we need to set this setting to `{Credo.Check.Design.TagTODO, false}`.

**Comment by wohali** [comment]
@van-mronov Probably a good idea.

**Comment by wohali** [comment]
@jjrodrig thank you!

**Pull request merged**